buggy_utils.py

```
@@ -0,0 +1,107 @@
+def read_config(filename):
+    base_path = "/etc/app/configs/"
+    with open(base_path + filename) as f:  # no sanitization
+        return f.read()
+
+
+# Memory leak - unbounded cache
+_cache = {}
+def cached_fetch(url):
+    if url not in _cache:
+        import urllib.request
+        _cache[url] = urllib.request.urlopen(url).read()
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)